File: 백준/Silver/1418. K－세준수/K－세준수.py
N = int(input())
K = int(input())

cnt = 0
for i in range(1, N + 1):
    sejun = True
    j = 2
    while i > 1:
        if j > K:  # 소인수가 K보다 큰 경우는 취급하지 않음
            sejun = False
            break
        if i % j == 0:  # 소인수 발견하면 나누기
            i /= j
        else:
            j += 1

    if sejun:
        cnt += 1

print(cnt)

File: 백준/Silver/1418. K－세준수/K－세준수.cc
#include <iostream>
using namespace std;

int main()
{
    int N, K;
    cin >> N >> K;

    int cnt = 0;
    for (int i = 1; i <= N; i++)
    {
        bool sejun = true;
        int num = i;
        int j = 2;
        while (num > 1)
        {
            if (j > K)
            {
                sejun = false;
                break;
            }
            if (num % j == 0)
            {
                num /= j;
            }
            else
            {
                j++;
            }
        }
        if (sejun == true)
        {
            cnt++;
        }
    }
    cout << cnt;
}